doc/admin-guide/files/ssl_server_name.config.en.rst

```diff
@@ -0,0 +1,124 @@
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+.. include:: ../../common.defs
+
+ssl_server_name.config
+**********************
+
+.. configfile:: ssl_server_name.config
+
+Description
+===========
+
+This file is used to configure aspects of TLS connection handling for both inbound and outbound
+connections. The configuration is driven the SNI values provided by the inbound connection.
+
+By default this is named :file:`ssl_server_name.config`. The file can be changed by settting
+:ts:cv:`proxy.config.ssl.servername.filename`.
+
+The configuration file is Lua based. After parsing and executing the Lua code, the global Lua variable
+``server_config`` is examined. Its value should be a list of tables. Each table is set of key / value
+pairs that create a configuration item.
+
+======================= ==============================================================================
+Key                     Meaning
+======================= ==============================================================================
+fqdn                    Fully Qualified Domain Name. This item is used if the SNI value matches this.
+
+verify_origin_server    One of the values :code:`NONE`, :code:`MODERATE`, or :code:`STRICT`.
+                        By default this is ::ts:cv:`proxy.config.ssl.client.verify.server`.
+
+verify_client           One of the values :code:`NONE`, :code:`MODERATE`, or :code:`STRICT`.
+                        By default this is :ts:cv:`proxy.config.ssl.client.certification_level`.
+
+client_cert             The client certificate to use for the outbound connection. If this is relative
+                        it is relative to the path in :ts:cv:`proxy.config.ssl.server.cert.path`.
+                        If not set :ts:cv:`proxy.config.ssl.client.cert.filename` is used.
+
+disable_h2              :code:`true` or :code:`false`. If :code:`false` then HTTP/2 is removed from
+                        the valid next protocol list. It is not an error to set this to :code:`false`
+                        for proxy ports on which HTTP/2 is not enabled.
+
+tunnel_route            Destination as an FQDN and port, separated by a colon '``:``'.
+======================= ==============================================================================
+
+Client verification, via ``verify_client``, correponds to setting
+:ts:cv:`proxy.config.ssl.client.certification_level` for this connection as noted below.
+
+:code:`NONE` -- ``0``
+   Do not request a client certificate.
+
+:code:`MODERATE` - ``1``
+   Request a client certificate and do verification if one is provide, but
+   do not enforce verification. A failure is logged but the connection is
```

Comment:
provided
@persiaAziz is the client verification enforced if the cert is provided?  As I recall this setting directly maps to an openssl setting.

Reply:
yes if the cert is provided, then the verification is enforced. (just verified again)